### ISSUE TITLE
Extending rest api functionality

### ISSUE THREAD

**Issue by 3210jr** [feathersjs/feathers-mongoose#121]
Hey @ekryski newbie questions here,

So I am (was) a die hard meteor fan, after a quick intro to feathers i think its even better. However, i'm stuck at a few places....

Using the feathers generate tool, and selecting `Api` and `rest api using socket.io`, and then selecting `mongodb` I get the app generated and ready to use rethink-mongoose.

Having come from  meteor, i am struggling to do any checking if the document already exists so that i don't insert duplicates, performing upserts, etc.

Can you please point me to any docs that address this.

In short, Where are all the db.find(), db.update(), db.... and stuff and how can i access this from the rest apis?

**Comment by daffl** [comment]
The database interactions in Feathers are done through the [service interface](https://docs.feathersjs.com/services/readme.html) and the [database adapter](https://docs.feathersjs.com/databases/readme.html) which all implement that interface.

Checking if an entry exists (and potential upserts) can be accomplished through [Hooks](https://docs.feathersjs.com/hooks/readme.html) which execute before or after any of the service methods runs. For example, a hook that checks for duplicate emails can look like this:

```js
const errors = require('feathers-errors');

app.service('myservice').before({
  create(hook) {
    return app.service('myservice').find({
      query: { email: hook.data.email }
    }, hook.params).then(results => {
      if(result.data.length !== 0) {
        throw new errors.Conflict(`Email ${hook.data.email} already exists`);
      }

      return hook;
    });
  }
});
```

**Comment by kaiquewdev** [comment]
Maybe you can get something on this section: https://docs.feathersjs.com/getting-started/scaffolding.html
